src/front/type_gen.rs

```diff
@@ -426,7 +424,7 @@ impl crate::Module {
                     (float_ty, int_ty, width as u32)
                 };
 
-                let mut type_name = "__frexp_result_".to_string();
+                let mut type_name = "_frexp_result_".to_string();
```

Comment:
We need to keep those as they are or else the user might be able to declare these types themselves and they will end up being equivalent.

Reply:
Np, do they need to make it to the output of `glsl` also? Because I’m afraid the updates to namer will remove it regardless.

Reply:
Also the update to `namer` has an effect on the other backends as well. 
Not sure if that was the desired effect or not (to have consistent names  across all of them). It just seemed like a convenient place to stick it. 

otherwise I could just remove it, and sanitize somewhere on the `glsl` backend only if that seems more appropriate.

There's still a handful of mixed usages of double and single underscore variables on some of the other backends not touched by this PR.

Reply:
Np, do they need to make it to the output of glsl also? Because I’m afraid the updates to namer will remove it regardless.

Also the update to namer has an effect on the other backends as well. Not sure if that was a desirable outcome or not. It just seemed like a convenient place to stick it.

There's still a handful of mixed usages of double and single underscore-prefixes on the other backends not touched by this PR... so it's hard to say if it has a negative impact or not.

Reply:
The namer is meant to be used by all textual backends so I guess it's fine. We can add per backend specific naming rules to it later if we decide it would be better.

> Np, do they need to make it to the output of glsl also? Because I’m afraid the updates to namer will remove it regardless.

Both the predeclared type and the user defined type will get renamed but since they will point to 2 different types in the arena they will get suffixed accordingly.